### PR TITLE
fix: We should not use blocking thread in frontend

### LINKED ISSUE #10704: batch: avoid spawning thread for every local execution
https://github.com/risingwavelabs/risingwave/blob/598e53190e6c8414060687d040f7b6854c58f2b3/src/frontend/src/scheduler/local.rs#L159-L160

I'm not sure whether this is acceptable. This somehow downgrades the execution into the thread model.

_Originally posted by @BugenZhao in https://github.com/risingwavelabs/risingwave/pull/7416#discussion_r1245275743_

### PR DESCRIPTION
**Pull request by liurenjie1024**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

Closes #10704 

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [x] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [x] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [x] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR contains user-facing changes.

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- Installation and deployment
- Connector (sources & sinks)
- SQL commands, functions, and operators
- RisingWave cluster configuration changes
- Other (please specify in the release note below)

### Release note

</details>

### CODE REVIEW

**Review by BugenZhao — COMMENTED** [review]
Looks like a quick fix, but I'm not sure whether it's ideal.

- How does `spawn(future)` and send to a channel compare with direct polling this stream in pgwire?
- If it's still possible to block the I/O driver, should we use a new dedicated runtime for these local tasks?

**Comment by chenzl25** [comment]
Let's use `select count(*) from generate_series(1, 500000000);` this statement to check whether the cancel can be handled timely

File: src/frontend/src/session.rs [hunk 1]
```diff
@@ -404,6 +412,21 @@ impl FrontendEnv {
...
     }
+
+    pub fn compute_runtime(&self) -> Arc<BackgroundShutdownRuntime> {
+        self.compute_runtime.clone()
+    }
+
+    fn create_compute_runtime() -> Arc<BackgroundShutdownRuntime> {
+        Arc::new(BackgroundShutdownRuntime::from(
+            Builder::new_multi_thread()
+                .worker_threads(4)
```
Comment (BugenZhao) [inline src/frontend/src/session.rs:423]: What about making this configurable?

**Review by BugenZhao — APPROVED** [review]
Rest LGTM

**Pull request merged**